### PR TITLE
[script][combat-trainer] Refactored Meraud commune in combat-trainer

### PR DESCRIPTION
**Pull request by MahtraDR**
Tested this and the prayer mat together. Both work reliably. Fingers crossed.

Also resolves https://github.com/rpherbig/dr-scripts/issues/5244.

### CODE REVIEW

File: combat-trainer.lic
Comment (rpherbig) [inline combat-trainer.lic:2670]: So this does return a bool, but it's only the result of putting away the holy water (not getting or sprinkling). Maybe that's ok, or maybe `sprinkle_holy_water` needs to be changed. What do you think?

File: combat-trainer.lic
Comment (rpherbig) [inline combat-trainer.lic:2658]: Can the trailing comma be removed?

File: combat-trainer.lic
Comment (rpherbig) [reply combat-trainer.lic:2670]: To put that differently, there's some redundancy/duplication in these three checks.

[DRCI.exists?](https://github.com/rpherbig/dr-scripts/blob/master/common-items.lic#L390) taps the water holder to make sure it's in the container.

[DRCTH.has_holy_water?](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L40) gets the water holder from the container (which is already a predicate function), checks if there is water inside, then puts the water holder away.

[DRCTH.sprinkle_holy_water](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L145) get the water holder form the container (again), calls `sprinkle?`, then puts the water holder away (again).

-----

I feel like what we really want here is "were we able to sprinkle the holy water?", which is _almost_ what `sprinkle_holy_water` does on its own. My only concern is that `sprinkle_holy_water` returns whether or not the item was put away, _not_ whether or not we could sprinkle the holy water.

Now, all of that said, `sprinkle_holy_water` is only used in two places in `commune.lic`, so I think what I'd like is to make a `sprinkle_holy_water?` which returns a more meaningful value and use it.

What do you think about that?

File: combat-trainer.lic
Comment (rpherbig) [inline combat-trainer.lic:2677]: There's an opportunity to combine all of this cleanup logic, something like:

```suggestion
unless DRCTH.sprinkle_holy_water?(@theurgy_supply_container, @water_holder, checkname) && DRCTH.wave_incense?(@theurgy_supply_container, @flint_lighter, checkname)
      DRC.message("Cannot sprinkle incense  and wave incense, skipping Meraud commune")
      @equipment_manager.wield_weapon?(game_state.weapon_name, game_state.weapon_skill)
      @training_abilities.delete('Meraud')
      return
  end
```

What do you think about that?

File: combat-trainer.lic
Comment (MahtraDR) [reply combat-trainer.lic:2670]: Actually, I found a bug in  [[DRCTH.has_holy_water?](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L40)](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L40) for the case where you can't get the water holder, but you can tap the water inside it. The function returns true, even though you can't get the holder to sprinkle yourself with. I'm finding a few weird odds and ends with common-theurgy, and more as I look.

Let me try and clean this up, but also work on common-theurgy on the side.

**Comment by rpherbig** [comment]
Looks good ✔️ 

Remind me to merge this in a few days once the common* changes have been picked up by everyone

File: combat-trainer.lic
Comment (MahtraDR) [reply combat-trainer.lic]: Good point. Moved it to the same line with the && operator.

**Review by rpherbig — APPROVED** [review]

**Comment by MahtraDR** [comment]
@rpherbig This should be good to merge.

I know I could, just checking first. :) If you're ok, I'll push the merge button to see what it looks like?

**Comment by rpherbig** [comment]
I appreciate you not taking the initiative to merge it, I like keeping that power for myself.

That said, go ahead and merge this one to see it :)

**Pull request merged**
